onnxruntime/python/_pybind_state.py

```diff
@@ -36,10 +43,18 @@
         if not os.path.isfile(os.path.join(cuda_bin_dir, f"cudnn64_{version_info.cudnn_version}.dll")):
             raise ImportError(f"cuDNN {version_info.cudnn_version} not installed in {cuda_bin_dir}.")
 
+        cupti_bin_dir = os.path.join(os.environ[cuda_env_variable], "extras", "CUPTI", "lib64")
+        if not os.path.isdir(cupti_bin_dir):
+            raise ImportError(f"cupti library path does not exist: {cupti_bin_dir}.")
+        elif not os.path.isfile(os.path.join(cupti_bin_dir, "cupti.lib")):
+            raise ImportError(f"cupti.lib is missing in: {cupti_bin_dir}.")
+
         if sys.version_info >= (3, 8):
             # Python 3.8 (and later) doesn't search system PATH when loading DLLs, so the CUDA location needs to be
             # specified explicitly using the new API introduced in Python 3.8.
             os.add_dll_directory(cuda_bin_dir)
+            os.add_dll_directory(cupti_bin_dir)
+            ExtendDllPath()
```

Comment:
Is this still needed if we've already added the dep for the cupti bin in the previous step?

Reply:
Yes we do - adding cupti alone does not silence the pipeline error complaining dll missing. This is because cupti introduced more dependeny only searchable from system path.

Reply:
Would this be a security issue? Can we add the specific directories to the path based on what cupti requires?